test/bar-markArea.html

```diff
@@ -168,7 +172,325 @@
+                  end: 60
+                },
+                series: [
+                  {
+                    type: 'bar',
+                    data: [[end - 1, 100, 22]],
+
+                    barWidth: 10,
+                    itemStyle: {
+                      opacity: 0.5
+                    },
+                    markArea: {
+                      itemStyle: {
+                        color: 'red',
+                        opacity: 0.5
+                      },
+                      data: [
+                        [
+                          {
+                            xAxis: 20
+                          },
+                          {
+                            xAxis: end - 10
+                          }
+                        ]
+                      ]
+                    }
+                  }
+                ]
+              };
+              var chart = testHelper.create(echarts, 'main5', {
+                  title: [
+                      'There should be a red markArea at the full grid area',
+                      'Fix #18130'
+                  ],
+                  option: option
+              });
+          });
+        </script>
+
+        <script>
+          require([
+              'echarts'
+          ], function (echarts) {
+              var option;
+              var end = 70;
+
+              var xData = [];
+              for (let i = 0; i < end; ++i) {
+                xData.push(i);
+              }
+
+              option = {
+                xAxis: [
+                  {
+                    data: xData,
+                    show: true
+                  }
+                ],
+                yAxis: [{ type: 'value', show: true, max: 100 }],
+                grid: [{}],
+                dataZoom: {
+                  type: 'slider',
+                  xAxisIndex: 0,
+                  startValue: 0,
+                  endValue: 61
+                },
+                series: [
+                  {
+                    type: 'bar',
+                    data: [[end - 1, 100, 22]],
+
+                    barWidth: 10,
+                    itemStyle: {
+                      opacity: 0.5
+                    },
+                    markArea: {
+                      itemStyle: {
+                        color: 'red',
+                        opacity: 0.5
+                      },
+                      data: [
+                        [
+                          {
+                            xAxis: 20
+                          },
+                          {
+                            xAxis: end - 10
+                          }
+                        ]
+                      ]
+                    }
+                  }
+                ]
+              };
+              var chart = testHelper.create(echarts, 'main6', {
+                  title: [
+                      'There should be a red markArea at 20~60. **Note the right should NOT be at the right boundary.**',
```

Comment:
I'm not sure if the last case is expected. 

<img src="https://user-images.githubusercontent.com/26999792/223015290-2f45d297-10f9-4de7-bab1-ebcac1da21f1.png" width="400">

1. Why the labels are neither aligned **_center_** nor aligned with ticks?
2. The red markArea seems **_over 60_**?

Reply:
1. I didn't change the logic of axis label. The logic of the alignment is that, when overlap, some ticks and labels are hidden, but the rest labels don't change position. For example, in the image below, the red tick is hidden because of overlapping, and the label `2` is aligned center according to the blue arrow range rather than orange arrow.

<img width="744" alt="image" src="https://user-images.githubusercontent.com/779050/223040742-37421657-e2d4-46c9-9824-11d7f768d703.png">

2. The following blue and green bars are at the x position of `2`. Similarly, the bar at the x position of `60` is as the yellow bar shows.

<img width="419" alt="image" src="https://user-images.githubusercontent.com/779050/223039836-65389bce-e121-4ac2-9f29-5096f6fd4372.png">

Reply:
Thanks for your detailed explaination!